tools/dailyjobs_query.py

```diff
@@ -0,0 +1,126 @@
+import MySQLdb
+import datetime
+from argparse import ArgumentParser
+import sys
+
+revisions_dict = {}
+database = MySQLdb.connect(host="localhost",
+                         user="root",
+                         passwd="root",
+                         db="ouija")
+cur = database.cursor()
+
+branches = ["mozilla-inbound", "fx-team"]
+platforms = ["linux", "osx", "win", "android"]
+
+
+def parse_args(argv=None):
+    parser = ArgumentParser()
+    parser.add_argument("-s", "--start_date",
+                        metavar="YYYY-MM-DD",
+                        default = datetime.date.today() - datetime.timedelta(days=1),
+                        dest="start_date",
+                        help="starting date."
+                        )
+
+    parser.add_argument("-e", "--end_date",
+                        metavar="YYYY-MM-DD",
+                        default = datetime.date.today(),
+                        dest="end_date",
+                        help="ending date."
+                        )
+
+    options = parser.parse_args(argv)
+    return options
+
+
+def updatedb(date, platform, branch, numpushes, numjobs, sumduration):
+    cur.execute('delete from dailyjobs where date="%s" and branch="%s" and platform="%s"' % (date, branch, platform))
+    query = 'insert into dailyjobs (date, platform, branch, numpushes, numjobs, sumduration) values ("{0}", "{1}", "{2}", {3}, {4}, {5})'
+    query = query.format(date, platform, branch, numpushes, numjobs, sumduration)
+    print query
+    cur.execute(query)
+
+
+def summarize(date, branch):
+    for platform in platforms:
+        numpushes = 0
+        numjobs = 0
+        sumduration = 0
+        for rev in revisions_dict:
+            if platform not in revisions_dict[rev]:
+                continue
+
+            value = revisions_dict[rev][platform]
+            if value[0] != date:
+                continue
+
+            numpushes += 1
+            numjobs += value[1]
+            sumduration += value[2]
+        print (date, platform, branch, numpushes, numjobs, sumduration)
+        updatedb(date, platform, branch, numpushes, numjobs, sumduration)
+
+
+def retrievedb(branch, date):
+    query = "select revision,count(result),sum(duration),platform from testjobs where branch='{0}' and date like '{1}%' group by revision,platform;"
+    query = query.format(branch, date)
+    print query
+    cur.execute(query)
+
+    for rows in cur.fetchall():
+        print "%s, %s" % (date, rows)
+        revision = rows[0]
+        jobs = int(rows[1])
+        duration = int(rows[2])
+        platform = rows[3]
+        found = False
+        for p in platforms:
+            if platform.startswith(p):
+                platform = p
+                found = True
+                break
+
+        # we can skip b2g/mulet or other platforms that don't match our core set
+        if not found:
```

Comment:
You could keep track of each not found platform in a set, then write a warning to the console or log once you've gone through all the data that says something like: "unrecognized platform: ..." This is optional, it might not end up being that useful.